### PR TITLE
Added custom asserts for checking models

### PR DESCRIPTION
**Pull request by hschilling**

### CODE REVIEW

File: openmdao/utils/assert_utils.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+    Raises
+    ------
+    AssertionError
+        If a subsystem of group is found to be using approximated partials.
+
+    """
+    groups_with_dict_jacobians = []
+    msg = 'The following groups use dictionary jacobians:\n'
+    for s in system.system_iter(include_self=include_self, recurse=recurse):
```
Comment (naylor-b) [inline openmdao/utils/assert_utils.py]: system_iter has a 'typ' arg that will cause it to yield only systems that are instances of the specified type, so you could just put typ=Group in there and get rid of the if isinstance(s, Group) block.

File: openmdao/utils/tests/test_assert_utils.py [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+from openmdao.api import Problem, ExplicitComponent, Group, IndepVarComp
+from openmdao.jacobians.assembled_jacobian import DenseJacobian
+from openmdao.test_suite.components.double_sellar import DoubleSellar
+from openmdao.test_suite.components.paraboloid import Paraboloid
+from openmdao.test_suite.components.sellar_feature import SellarNoDerivativesCS
+from openmdao.utils.assert_utils import assert_check_partials, assert_no_approx_partials, assert_no_dict_jacobians
+
+
+class TestAssertUtils(unittest.TestCase):
+    # def setUp(self):
```
Comment (kmarsteller) [inline openmdao/utils/tests/test_assert_utils.py]: Keeping this around for a rainy day? :)

Comment (swryan) [reply openmdao/utils/tests/test_assert_utils.py]: destroy the evidence!

File: openmdao/docs/debug_build_docs.py [hunk 3]
```diff
@@ -0,0 +1,7 @@
+if __name__ == '__main__':
```
[the remark this answers is not in the document]
Comment (hschilling) [reply openmdao/docs/debug_build_docs.py]: done

File: openmdao/utils/tests/test_assert_utils.py (hunk 2 above)
Comment (hschilling) [reply openmdao/utils/tests/test_assert_utils.py]: Sorry. Done

File: openmdao/utils/assert_utils.py (hunk 1 above)
Comment (hschilling) [reply openmdao/utils/assert_utils.py]: thanks for the tip. Done

**Pull request merged**